unstable/scripts/exconverter/main.go

```diff
@@ -0,0 +1,168 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+func main() {
+	keep := flag.Bool("keep", false, "set to true to keep the temp file")
+	flag.Parse()
+
+	tmpdir, err := os.MkdirTemp("", "example-coverage-metrics")
+	noerr(err)
+	if !*keep {
+		defer os.RemoveAll(tmpdir)
+	}
+
+	baselinedir := filepath.Join(tmpdir, "baseline")
+	experimentaldir := filepath.Join(tmpdir, "experimental")
+
+	os.Setenv("GOWORK", "off")
+
+	os.Setenv("COVERAGE_OUTPUT_DIR", experimentaldir)
+	tfgen(1)
+
+	os.Setenv("COVERAGE_OUTPUT_DIR", baselinedir)
+	tfgen(0)
+
+	baselinestats := readstats(baselinedir)
+	fmt.Println("Baseline    ", len(baselinestats.exampleByHCL))
+	fmt.Println(baselinestats.shortSummary)
+
+	experimentalstats := readstats(experimentaldir)
+	fmt.Println("Experimental", len(experimentalstats.exampleByHCL))
+	fmt.Println(experimentalstats.shortSummary)
+
+	for _, e := range baselinestats.newlyFailing(experimentalstats) {
+		fmt.Printf("Example started failing: %v\n", e.ExampleName)
+		for lang, cr := range e.ConversionResults {
+			fmt.Println(lang, cr.FailureSeverity, cr.FailureInfo)
+		}
+	}
+
+	for _, e := range baselinestats.dropped(experimentalstats) {
+		fmt.Printf("Example dropped: %v\n", e.ExampleName)
+	}
+
+	fmt.Println("DONE")
+}
+
+type stats struct {
+	shortSummary string
+	exampleByHCL map[string]flattenedExample
+}
+
+func (s stats) dropped(new stats) []flattenedExample {
+	out := []flattenedExample{}
+	for k, v := range s.exampleByHCL {
+		if _, ok := new.exampleByHCL[k]; !ok {
+			out = append(out, v)
+		}
+	}
+	return out
+}
+
+func (s stats) newlyFailing(new stats) []flattenedExample {
+	out := []flattenedExample{}
+	for k, v := range s.exampleByHCL {
+		if vn, ok := new.exampleByHCL[k]; ok {
+			for lang, newCR := range vn.ConversionResults {
+				if oldCR, ok := v.ConversionResults[lang]; ok {
+					if oldCR.FailureSeverity < newCR.FailureSeverity &&
+						newCR.FailureSeverity > warning {
+						out = append(out, vn)
+					}
+				}
+			}
+		}
+	}
+	return out
+}
+
+type flattenedExample struct {
+	ExampleName       string
+	OriginalHCL       string `json:"OriginalHCL,omitempty"`
+	ConversionResults map[string]*languageConversionResult
+}
+
+const (
+	success = 0
+	warning = 1
+	failure = 2
+	fatal   = 3
+)
+
+type languageConversionResult struct {
+	FailureSeverity  int    // This conversion's outcome: [Success, Warning, Failure, Fatal]
+	FailureInfo      string // Additional in-depth information
+	Program          string // Converted program
+	TranslationCount int
+}
+
+func readstats(dir string) stats {
+	f, err := os.Open(filepath.Join(dir, "byExample.json"))
+	noerr(err)
+	defer func() {
+		noerr(f.Close())
+	}()
+
+	s := stats{
+		exampleByHCL: map[string]flattenedExample{},
+	}
+
+	dec := json.NewDecoder(f)
+	for {
+		var entry flattenedExample
+		err := dec.Decode(&entry)
+		if err != nil || entry.ExampleName == "" {
+			break
```

Comment:
It feels like this should be a `continue`. Why are we stopping on the first error?
Why don't we report the error?